lib/facility.rb

```diff
@@ -1,14 +1,70 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader   :name,
+                :address,
+                :phone,
+                :services,
+                :registered_vehicles
+  attr_accessor :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def register_vehicle(vehicle)
+    regular_vehicle_fee = 100
+    ev_fee = 200
+    antique_vehicle_fee = 25
+    vehicle.registration_date = Date.today
+
+    if vehicle.engine == :ev && @services.include?('Vehicle Registration')
+      vehicle.plate_type = :ev
+      @collected_fees += ev_fee
+      registered_vehicles << vehicle
+    elsif vehicle.antique? == true && @services.include?('Vehicle Registration')
+      vehicle.plate_type = :antique
+      @collected_fees += antique_vehicle_fee
+      registered_vehicles << vehicle
+    elsif @services.include?('Vehicle Registration')
+      vehicle.plate_type = :regular
+      @collected_fees += regular_vehicle_fee
+      registered_vehicles << vehicle
+    else
+      nil
+    end
+  end
+
+  def add_service(service)
     @services << service
   end
+
+  def administer_written_test(registrant)
+    if (registrant.permit?) && (registrant.age > 15) && (@services.include?('Written Test'))
```

Comment:
In your submission form you asked:
> Besides refactoring the code, what's the best way to separate multiple lines from one.  There are good examples of what I'm talking about in my Facility class...Line 46, Line 54, and Line 63

I'm not totally sure what you mean by this. Can you say a little more about what you're asking? If you'd like to jump on a call to talk through if that's easily, let me know. Happy to do so.